src/Components/Web/src/Forms/InputRadio.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// An input component used for selecting a value from a group of choices.
+    /// </summary>
+    public class InputRadio<TValue> : InputChoice<TValue>
```

Comment:
When components are reused then relying on a unique name becomes difficult.

Could you consider a wrapping `<InputRadioGroup>` which will

A: Let the developer specify a `Name`
B: Generate a GUID as a name when none is specified

It would pass itself down as a CascadingValue so your child components can pick up the name. If the developer sets a `name` explicitly on InputRadio then use that, if not the try the cascaded object's name, and if that isn't there then throw an exception. 

I hope I explained that well enough :) 

What you have is very good, but I think we need a bit of flexibility added for it to work nicely when we place multiple instances on a page of a reusable component that has radio inputs (without the dev having to pass down a unique name to our reusable component) - especially when that component is created in a foreach loop.

Reply:
This is a great suggestion, I've now added an optional `<InputRadioGroup>` component.

Reply:
Thank you for doing this. You've really done an excellent job!